server/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -58,20 +58,9 @@ object CORS {
    * Currently, you cannot make permissions depend on request details
    */
   def apply(service: HttpService, config: CORSConfig = DefaultCORSConfig): HttpService = Service.lift { req =>
-
-    def options(origin: Header, acrm: Header): HttpService =
-      (service |+| ok).map {
-        case resp: Response =>
-          if (resp.status.isSuccess)
-            corsHeaders(origin.value, acrm.value)(resp)
-          else {
-            logger.info(s"CORS headers would have been allowed for ${req.method} ${req.uri}")
-            resp
-          }
-        case Pass =>
-          logger.warn("Unexpected Pass in CORS. This is probably a bug.")
-          Pass
-      }
+    
+    //Always reply with success for an OPTIONS request. The headers to send back are determined by the config.
+    def options(origin: Header, acrm: Header): HttpService = ok.map(corsHeaders(origin.value, acrm.value)(_))
```

Comment:
I'm compiling in my head, but I think this can return a simple `Response`, and then where this is called, avoid the application of the request.  The request doesn't matter anymore.